dotnet: nixfmt output of nuget-to-nix

corngood opened the pull request:
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

corngood committed:
dotnet: nixfmt output of nuget-to-nix

corngood commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 361579`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 5 packages built:</summary>
  <ul>
    <li>dotnet-outdated</li>
    <li>juniper</li>
    <li>netcoredbg</li>
    <li>tagger</li>
    <li>yafc-ce</li>
  </ul>
</details>

MattSturgeon commented:
Related discussion:
- #325053
- #358025

MattSturgeon commented on pkgs/build-support/dotnet/add-nuget-deps/default.nix:
```diff
@@ -67,6 +68,7 @@ attrs
             isExecutable = true;
             inherit cacert;
             nugetToNix = nuget-to-nix;
+            nixfmt = nixfmt-rfc-style;
```
Not really relevant to this PR, but rather than substituting these in, would it make sense to instead set PATH using `lib.mkBinPath [ nuget-to-nix nixfmt-rfc-style ]`?

MattSturgeon reviewed (approved):
No issues from me. I think JSON lockfiles will be better in the long term, but I'm fine with this too.

I also believe concerns raised by others would be reassured if this is treated as a stopgap solution?

corngood commented:
We could also do this with or without actually updating the lockfiles.  I have them in here now, which was useful to check the rebuild count, etc.

corngood committed:
dotnet: use fallback target packages from current binary sdk

This was using the bootstrap SDK, which doesn't match the version of the
source-built SDK.  This will still be a problem if the binary and
source-built SDKs don't have the same version.

corngood committed:
Revert "scarab: remove nuget patch"

This reverts commit 7cd9f932236ead6a8af90c3618406845b3111a56.

corngood commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 361579`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 5 packages built:</summary>
  <ul>
    <li>dotnet-outdated</li>
    <li>juniper</li>
    <li>netcoredbg</li>
    <li>scarab</li>
    <li>yafc-ce</li>
  </ul>
</details>

gepbird commented on pkgs/by-name/sc/scarab/package.nix:
```diff
@@ -31,6 +31,12 @@ buildDotnetModule rec {
       --replace-fail 'net6.0' 'net8.0'
   '';
 
+  preConfigureNuGet = ''
```
Thanks for catching that, most likely I was looking at another package that recently added the this nuget source upstrea,, apologies for the lack of testing.

corngood replied:
No worries.  We don't really have tools in place for making sure `fetch-deps` continues to work for a package.

MattSturgeon commented:
> We could also do this with or without actually updating the lockfiles. I have them in here now, which was useful to check the rebuild count, etc.

I think my preference would be _not_ to do a treewide reformat yet. CI only requires formatting for new files (and already compliant files).

The [nixpkgs-review build](https://github.com/NixOS/nixpkgs/pull/361579#issuecomment-2520703524) _with_ the reformat is reassuring either way.

Depending on timing, there's a chance a JSON-based solution may be ready before many of these packages end up bumping their lockfiles.

Not that this is important either way. I have no _objection_ to a reformat now.

corngood commented:
> I think my preference would be not to do a treewide reformat yet. CI only requires formatting for new files (and already compliant files).

Done.  I don't think there's really a good argument for regenerating, other than to prove that it can be done.
